contracts/src/LightClient.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+import { BN254 } from "bn254/BN254.sol";
+import { IPlonkVerifier } from "./interfaces/IPlonkVerifier.sol";
+import { IStakeTable } from "./interfaces/IStakeTable.sol";
+import { PlonkVerifier } from "./libraries/PlonkVerifier.sol";
+
+// TODO: replace these, for now WIP and test only
+import { VkTest } from "../test/stubs/Transfer1In2Out24DepthVk.sol";
+
+/// @notice A light client for HotShot consensus. Keeping track of its finalized states in safe,
+/// authenticated ways.
+contract LightClient {
+    // === Constants ===
+    //
+    /// @notice System parameter: number of blocks per epoch
+    uint32 public constant BLOCKS_PER_EPOCH = 20_000;
+
+    // === Storage ===
+    //
+    /// @notice genesis block commitment
+    LightClientState public genesisState;
+    /// @notice global storage of the finalized HotShot's light client state
+    LightClientState public finalizedState;
+    /// @notice current (finalized) epoch number
+    uint64 public currentEpoch;
+    /// @notice address of the stake table contract
+    IStakeTable internal stakeTable;
+
+    // === Data Structure ===
+    //
+    /// @notice The finalized HotShot state (as the digest of the entire HotShot state)
+    /// @param viewNum The latest view number of the finalized HotShot chain
+    /// @param blockHeight The block height of the latest finalized block
+    /// @param blockComm The block commitment (type: BN254::ScalarField)
+    /// @param feeLedgerComm The commitment to the fee ledger state (type: BN254::ScalarField)
```

Comment:
Same here, why field element vs `bytes32`?

Reply:
Here is the `Field::from_le_bytes(actual_comm: bytes32)` result that is also used to generate the proof.